docs/architecture/adr-050-sign-mode-textual.md

```diff
@@ -0,0 +1,504 @@
+                "from": "cosmos1...stu",
+                "to": "cosmos1...vwx",
+                "amount": [
+                  {
+                    "denom": "uatom",
+                    "amount": 30000000
+                  }
+                ]
+              },
+              {
+                "@type": "/cosmos.bank.v1beta1.MsgSend",
+                "from": "cosmos1...abc",
+                "to": "cosmos1...def",
+                "amount": [
+                  {
+                    "denom": "uatom",
+                    "amount": 40000000
+                  }
+                ]
+              }
+            ]
+          }
+        ],
+        "initial_deposit": [
+          {
+            "denom": "atom",
+            "amount": 100.01
+          }
+        ]
+      }
+    ]
+  },
+  "auth_info": {
+    "signer_infos": [
+      {
+        "public_key": "iQ...==",
+        "mode_info": { "single": { "mode": "SIGN_MODE_TEXTUAL" } },
+        "sequence": 2
+      },
+      {
+        "public_key": "iR...==",
+        "mode_info": { "single": { "mode": "SIGN_MODE_DIRECT" } },
+        "sequence": 42
+      }
+    ],
+    "fee": {
+      "amount": [
+        {
+          "denom": "atom",
+          "amount": 0.002
+        }
+      ],
+      "gas_limit": 100000
+    }
+  },
+  // Additional SignerData.
+  "chain_id": "simapp-1",
+  "account_number": 10
+}
+}
+```
+
+SIGN_MODE_TEXTUAL for 1st signer `cosmos1...abc`:
+
+```
+Chain ID: simapp-1
+Account number: 10
+*Public Key: iQ...==
+Sequence: 2
+This transaction has 2 messages:
+Message (1/2): bank v1beta1 send coins
+From: cosmos1...abc
+To: cosmos1...def
+Amount: 10 atom
+Message (2/2): gov v1 submit proposal
+Messages: 2 Messages
+> Message (1/2): bank v1beta1 send coins
+> From: cosmos1...jkl
+> To: cosmos1...mno
+> Amount: 20 atom
+> Message (2/2): authz v1beta exec
+> Grantee: cosmos1...pqr
+> Msgs: 2 Msgs
+>> Msg (1/2): bank v1beta1 send coins
+>> From: cosmos1...stu
+>> To: cosmos1...vwx
+>> Amount: 30 atom
+>> Msg (2/2): bank v1beta1 send coins
+>> From: cosmos1...abc
+>> To: cosmos1...def
+>> Amount: 40 atom
+> End of Msgs
+End of transaction messages
+Proposer: cosmos1...ghi
+Initial Deposit: 100.01 atom
+End of transaction messages
+Fee: 0.002 atom
+*Gas: 100'000
+*This transaction has 1 other signer:
+*Signer (2/2):
+*Public Key: iR...==
+*Sign mode: Direct
+*Sequence: 42
+*End of other signers
+```
+
+## Consequences
+
+### Backwards Compatibility
+
+SIGN_MODE_TEXTUAL is purely additive, and doesn't break any backwards compatibility with other sign modes.
+
+### Positive
+
+- Human-friendly way of signing on hardware devices.
+- Once SIGN_MODE_TEXTUAL is shipped, SIGN_MODE_LEGACY_AMINO_JSON can be deprecated and removed. On the longer term, once the ecosystem has totally migrated, Amino can be totally removed.
+
+### Negative
+
+- If the transaction is complex, the string array can be arbitrarily long, and some users might just skip some screens and blind sign.
```

Comment:
I think this is neutral.... there is no solution for long transactions if we want to present all key information.